tests/lsdb/catalog/test_join.py

```diff
@@ -31,3 +32,40 @@ def test_join_wrong_columns(small_sky_catalog, small_sky_order1_catalog):
 def test_join_wrong_suffixes(small_sky_catalog, small_sky_order1_catalog):
     with pytest.raises(ValueError):
         small_sky_catalog.join(small_sky_order1_catalog, left_on="id", right_on="id", suffixes=("wrong",))
+
+
+def test_join_association(small_sky_catalog, small_sky_xmatch_catalog, small_sky_to_xmatch_catalog):
+    suffixes = ("_a", "_b")
+    joined = small_sky_catalog.join(small_sky_xmatch_catalog,
+                                    through=small_sky_to_xmatch_catalog,
+                                    suffixes=suffixes)
+    assert joined._ddf.npartitions == len(small_sky_to_xmatch_catalog.hc_structure.join_info.data_frame)
+    joined_data = joined.compute()
+    association_data = small_sky_to_xmatch_catalog.compute()
+    assert len(joined_data) == len(association_data)
+    for _, row in association_data.iterrows():
+        left_col = small_sky_to_xmatch_catalog.hc_structure.catalog_info.primary_column + suffixes[0]
+        right_col = small_sky_to_xmatch_catalog.hc_structure.catalog_info.join_column + suffixes[1]
+        left_id = row[small_sky_to_xmatch_catalog.hc_structure.catalog_info.primary_column_association]
+        right_id = row[small_sky_to_xmatch_catalog.hc_structure.catalog_info.join_column_association]
+        joined_row = joined_data.query(f"{left_col} == {left_id} & {right_col} == {right_id}")
+        assert len(joined_row) == 1
+        small_sky_col = small_sky_to_xmatch_catalog.hc_structure.catalog_info.primary_column
+        left_index = small_sky_catalog.compute().query(f"{small_sky_col}=={left_id}").index
+        assert joined_row.index == left_index
```

Comment:
Can we also please check that the columns in the association data are the columns we expect?